### PR TITLE
Interfaces 422 wagmi wallet connector

### PR DESCRIPTION
**Pull request by sumbat-ssvlabs**

### CODE REVIEW

File: src/app/Routes/FaucetRoutes.tsx [hunk 1]
```diff
@@ -17,7 +15,7 @@ const FaucetRoutes: any = () => {
 
   return (
     <Layout>
-      <AppBar excludeNetworks={[MAINNET_NETWORK_ID]} />
```
Comment (axelrod-blox) [inline src/app/Routes/FaucetRoutes.tsx:20]: make sure we don't have mainnet button on faucet app

File: src/app/Routes/DistributionRoutes.tsx [hunk 2]
```diff
@@ -16,12 +14,12 @@ const DistributionRoutes: any = () => {
 
   return (
     <Layout>
-      <AppBar excludeNetworks={[HOLESKY_NETWORK_ID]} />
```
Comment (axelrod-blox) [inline src/app/Routes/DistributionRoutes.tsx:19]: make sure we don't have holesky button on faucet app

File: src/app/components/applications/SSV/MyAccount/components/Withdraw/Withdraw.tsx [hunk 3]
```diff
@@ -1,20 +1,20 @@
...
-import { useStyles } from '~app/components/applications/SSV/MyAccount/components/Withdraw/Withdraw.styles';
-import ProcessStore from '~app/common/stores/applications/SsvWeb/Process.store';
-import { fromWei, toDecimalNumber } from '~root/services/conversions.service';
 import { getClusterBalance } from '~root/services/cluster.service';
-import { SingleOperator, SingleCluster } from '~app/model/processes.model';
-import OperatorFlow from './OperatorFlow';
+import { fromWei, toDecimalNumber } from '~root/services/conversions.service';
 import ClusterFlow from './ClusterFlow';
-import { getNetworkFeeAndLiquidationCollateral } from '~app/redux/network.slice';
+import OperatorFlow from './OperatorFlow';
```
Comment (axelrod-blox) [inline src/app/components/applications/SSV/MyAccount/components/Withdraw/Withdraw.tsx:17]: absolute path

File: src/app/components/applications/SSV/MyAccount/components/Withdraw/Withdraw.tsx [hunk 4]
```diff
@@ -1,20 +1,20 @@
...
-import BorderScreen from '~app/components/common/BorderScreen';
-import NewWhiteWrapper from '~app/components/common/NewWhiteWrapper/NewWhiteWrapper';
-import { useStyles } from '~app/components/applications/SSV/MyAccount/components/Withdraw/Withdraw.styles';
-import ProcessStore from '~app/common/stores/applications/SsvWeb/Process.store';
-import { fromWei, toDecimalNumber } from '~root/services/conversions.service';
 import { getClusterBalance } from '~root/services/cluster.service';
-import { SingleOperator, SingleCluster } from '~app/model/processes.model';
-import OperatorFlow from './OperatorFlow';
+import { fromWei, toDecimalNumber } from '~root/services/conversions.service';
 import ClusterFlow from './ClusterFlow';
```
Comment (axelrod-blox) [inline src/app/components/applications/SSV/MyAccount/components/Withdraw/Withdraw.tsx:16]: absolute path

File: src/app/components/applications/SSV/RegisterValidatorHome/components/ImportFile/flows/KeyShares.tsx [hunk 5]
```diff
@@ -1,53 +1,49 @@
...
-import OperatorData
-  from '~app/components/applications/SSV/RegisterValidatorHome/components/ImportFile/flows/Operator/OperatorData';
+import WarningBox from '~app/components/common/WarningBox';
+import { useStores } from '~app/hooks/useStores';
 import validatorRegistrationFlow, { EBulkMode, EValidatorFlowAction } from '~app/hooks/useValidatorRegistrationFlow';
-import ValidatorList
-  from '~app/components/applications/SSV/RegisterValidatorHome/components/ImportFile/flows/ValidatorList/ValidatorList';
-import ValidatorCounter
-  from '~app/components/applications/SSV/RegisterValidatorHome/components/ImportFile/flows/ValidatorList/ValidatorCounter';
+
```
Comment (axelrod-blox) [inline src/app/components/applications/SSV/RegisterValidatorHome/components/ImportFile/flows/KeyShares.tsx:23]: empty field?

File: src/app/components/common/ConnectWalletDialog/ConnectWalletDialog.tsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+import { FC } from 'react';
+import { useLocalStorage } from 'react-use';
+import { Connector, UseAccountEffectParameters, useAccountEffect, useConnect } from 'wagmi';
+import { Checkbox } from '~app/components/ui/checkbox';
+import { Dialog, DialogContent, DialogHeader, DialogTitle } from '~app/components/ui/dialog';
+import { useAppSelector } from '~app/hooks/redux.hook';
+import { getIsDarkMode } from '~app/redux/appState.slice';
+import { cn } from '~lib/utils/tailwind';
+import { WalletOption } from './WalletOption';
+import { WalletOptionCard } from './WalletOptionCard';
```
Comment (axelrod-blox) [inline src/app/components/common/ConnectWalletDialog/ConnectWalletDialog.tsx:11]: absolute path line 10, 11

File: src/app/hooks/useClusters.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+//         page: 1,
+//         per_page: 7,
+//         total: 0
+//       }
+//     },
+//     refetchOnWindowFocus: false,
+//     refetchOnMount: false,
+//     enabled: Boolean(account.address)
+//   });
+// };
```
Comment (axelrod-blox) [inline src/app/hooks/useClusters.ts:26]: remove if unused

File: src/index.tsx [hunk 8]
```diff
@@ -1,26 +1,37 @@
...
-import { rootStore } from '~root/stores';
-import * as serviceWorker from '~root/serviceWorker';
-import GTMFrame from '~lib/analytics/GoogleTag/components/GTMFrame';
 import { store } from '~app/store';
-import { Provider as RdProvider } from 'react-redux';
+import GTMFrame from '~lib/analytics/GoogleTag/components/GTMFrame';
+import * as serviceWorker from '~root/serviceWorker';
+import { rootStore } from '~root/stores';
+import { RainbowKitProvider } from './RainbowKitProvider';
+import { config } from './wagmi/config';
```
Comment (axelrod-blox) [inline src/index.tsx:13]: absolute paths

File: src/providers/networkInfo.provider.ts [hunk 9]
```diff
@@ -44,6 +44,13 @@ const LINKS = {
 
 const rawNetworksData: string = import.meta.env.VITE_SSV_NETWORKS as string;
 const networks: NetworkInfo[] = JSON.parse(rawNetworksData);
+// const networksMap = networks.reduce(
```
Comment (axelrod-blox) [inline src/providers/networkInfo.provider.ts:47]: let's delete it

**Review by axelrod-blox — APPROVED** [review]

File: src/app/Routes/FaucetRoutes.tsx (hunk 1 above)
Comment (sumbat-ssvlabs) [reply src/app/Routes/FaucetRoutes.tsx:20]: Now we choose the networks for each app in the Wagmi config

```typescript
const appChains: Record<typeof app, [Chain, ...Chain[]]> = {
  ssvweb: [mainnet, holesky],
  distribution: [mainnet],
  faucet: [holesky],
};
```

File: src/app/Routes/DistributionRoutes.tsx (hunk 2 above)
Comment (sumbat-ssvlabs) [reply src/app/Routes/DistributionRoutes.tsx:19]: "Now we choose the networks for each app in the Wagmi config"

File: src/app/components/applications/SSV/MyAccount/components/Validator/SingleCluster/SingleCluster.tsx [hunk 10]
```diff
@@ -1,62 +1,59 @@
...
-import { getIsDarkMode } from '~app/redux/appState.slice';
+import { useAppDispatch, useAppSelector } from '~app/hooks/redux.hook';
+import { useStores } from '~app/hooks/useStores';
+import useValidatorRegistrationFlow from '~app/hooks/useValidatorRegistrationFlow';
+import { SingleCluster as SingleClusterProcess } from '~app/model/processes.model';
 import { getSelectedCluster, setSelectedClusterId } from '~app/redux/account.slice';
+import { getIsDarkMode } from '~app/redux/appState.slice';
+import { getAccountAddress } from '~app/redux/wallet.slice';
+import { isEqualsAddresses } from '~lib/utils/strings';
+import { useStyles } from './SingleCluster.styles';
```
Comment (axelrod-blox) [inline src/app/components/applications/SSV/MyAccount/components/Validator/SingleCluster/SingleCluster.tsx:24]: use absolute path

File: src/app/components/applications/SSV/MyAccount/components/Validator/SingleCluster/SingleCluster.tsx [hunk 11]
```diff
@@ -71,16 +68,16 @@ const SingleCluster = () => {
   const isDarkMode = useAppSelector(getIsDarkMode);
   const accountAddress = useAppSelector(getAccountAddress);
 
-  const hasPrivateOperator = cluster.operators.some((operator: any) => operator.address_whitelist && !isEqualsAddresses(operator.address_whitelist, accountAddress) && operator.address_whitelist !== config.GLOBAL_VARIABLE.DEFAULT_ADDRESS_WHITELIST);
+  console.log('cluster:', cluster);
```
Comment (axelrod-blox) [inline src/app/components/applications/SSV/MyAccount/components/Validator/SingleCluster/SingleCluster.tsx:71]: unnecessary console.log

File: src/app/components/common/AppBar/AppBar.tsx [hunk 12]
```diff
@@ -1,143 +1,140 @@
...
-    const isLoading = useAppSelector(getIsLoading);
-    const strategyRedirect = useAppSelector(getStrategyRedirect);
-    const hasOperatorsOrValidators = strategyRedirect === config.routes.SSV.MY_ACCOUNT.CLUSTER_DASHBOARD;
+const AppBar = ({ buttons }: { buttons?: Button[] }) => {
+  const navigate = useNavigate();
+  const wrapperRef = useRef(null);
+  const buttonsRef = useRef(null);
+  const [menuBar, openMenuBar] = useState(false);
+  // const isDistribution = applicationStore.strategyName === 'distribution';
+  // @ts-ignore
```
Comment (axelrod-blox) [inline src/app/components/common/AppBar/AppBar.tsx:26]: I think we can remove this ts-ignore, and line 25 as well

File: src/app/components/common/NewWhiteWrapper/components/ValidatorsFlow/ValidatorsFlow.tsx [hunk 13]
```diff
@@ -1,26 +1,22 @@
...
-import { SingleCluster } from '~app/model/processes.model';
 import { useAppSelector } from '~app/hooks/redux.hook';
+import { useStores } from '~app/hooks/useStores';
+import { SingleCluster } from '~app/model/processes.model';
 import { getAccountAddress } from '~app/redux/wallet.slice';
-import { useDispatch } from 'react-redux';
-import { setSelectedClusterId } from '~app/redux/account.slice';
+import { longStringShorten } from '~lib/utils/strings';
+import { getClusterHash } from '~root/services/cluster.service';
+import { useStyles } from '../../NewWhiteWrapper.styles';
```
Comment (axelrod-blox) [inline src/app/components/common/NewWhiteWrapper/components/ValidatorsFlow/ValidatorsFlow.tsx:11]: absolute path

File: src/app/hooks/useFetchWalletBalance.ts [hunk 14]
```diff
@@ -16,7 +16,7 @@ const useFetchWalletBalance = () => {
       dispatch(setIsShowSsvLoader(false));
     };
     fetchWalletBalance();
-  }, [accountAddress, dispatch]);
+  }, [/* accountAddress */ dispatch]);
```
Comment (axelrod-blox) [inline src/app/hooks/useFetchWalletBalance.ts:19]: remove if unused

**Review by IlyaVi — APPROVED** [review]

File: src/app/components/applications/SSV/WalletPopUp/WalletPopUp.tsx [hunk 15]
```diff
@@ -1,45 +1,46 @@
-
-import Grid from '@mui/material/Grid';
 import Dialog from '@mui/material/Dialog';
+import Grid from '@mui/material/Grid';
```
Comment (IlyaVi) [inline src/app/components/applications/SSV/WalletPopUp/WalletPopUp.tsx:2]: delete

**Review by IlyaVi — APPROVED** [review]

**Pull request merged**